### PR TITLE
WFLY-5970 Plain JMS clients don't fallback

### PR DESCRIPTION
**Pull request by treblereel**
(cherry picked from commit 1f40a86fd7134d8b2211bdad4d7028e24b6003bd)

url: https://issues.jboss.org/browse/WFLY-5970

### CODE REVIEW

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/TransportConfigOperationHandlers.java [hunk 1]
```diff
@@ -296,7 +295,7 @@ static void processAcceptors(final OperationContext context, final Configuration
      * @param bindings      the referenced socket bindings
      * @throws OperationFailedException
      */
-    static void processConnectors(final OperationContext context, final Configuration configuration, final ModelNode params, final Set<String> bindings) throws OperationFailedException {
+    public static void processConnectors(final OperationContext context, final Configuration configuration, final ModelNode params, final Set<String> bindings) throws OperationFailedException {
```
Comment (bstansberry) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/TransportConfigOperationHandlers.java]: This visibility change is not needed.

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/TransportConfigOperationHandlers.java [hunk 2]
```diff
@@ -310,6 +309,13 @@ static void processConnectors(final OperationContext context, final Configuratio
                     parameters.put(GenericTransportDefinition.SOCKET_BINDING.getName(), socketBinding.asString());
                 }
                 final String clazz = FACTORY_CLASS.resolveModelAttribute(context, config).asString();
+                //we treat it as remote_connector
+                ModelNode bindingNode = config.get(GenericTransportDefinition.SOCKET_BINDING.getName());
+                if (bindingNode != null && bindingNode.isDefined()) {
+                    final String binding = bindingNode.asString();
+                    parameters.put(GenericTransportDefinition.SOCKET_BINDING.getName(), binding);
+                    bindings.add(binding);
+                 }
```
Comment (bstansberry) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/TransportConfigOperationHandlers.java]: This has the same effect as the logic at L305-310.

**Comment by treblereel** [comment]
@bstansberry  let me check it tomorrow ...

**Comment by treblereel** [comment]
retest this please

**Comment by treblereel** [comment]
@bstansberry fixed

**Comment by vpakan** [comment]
@bstansberry please could you merge this PR since it is upstream fix for BZ1258304 and it's included in next EAP 6.4. payload?

**Comment by bstansberry** [comment]
@vpakan It needs an equivalent PR against the jboss-eap7/7.x branch. For background I just forwarded a recent thread about this to the SET mailing list.

**Comment by treblereel** [comment]
@bstansberry  this is PR to jboss-eap7/7.x  you asked https://github.com/jbossas/jboss-eap7/pull/748

**Comment by bstansberry** [comment]
@vpakan @treblereel Sorry, I should have looked more closely again at the PR before asking for 7.x.

This PR is functionally a no-op. It just does the same thing as the existing code in a slightly less correct way, e.g. not using resolveModelAttribute. I commented previously on one location where that was the case but every other change has no functional effect AFAICT. If I'm missing something please point it out.

@jmesnil is WFLY-5970 a duplicate for some other issue? 

Perhaps there is a EAP 6 issue that was dealt with in WildFly/EAP 7 long ago and there is no need for an upstream change?  WFLY-5970 says it's a bug in 10.0.0.Final though and this class was in its current form long before that.

**Comment by jmesnil** [comment]
@bstansberry WFLY-5970 is a dup of https://issues.jboss.org/browse/WFLY-5394 that was fixed in WFLY 10.0.0.CR2.
This PR can be closed, I've resolved the WFLY-5970 issue as duplicate.

**Comment by bstansberry** [comment]
Thanks @jmesnil.

@treblereel @vpakan I'll close this and the 7.x one. As for the upstream-first requirement for a 6.4.x fix, you can link to this discussion as evidence that this is already handled upstream.

**Pull request closed without merging**